main.tf

```diff
@@ -256,9 +256,8 @@ data "aws_ami" "docker-machine" {
 }
 
 resource "aws_autoscaling_group" "gitlab_runner_instance" {
-  name                = "${var.environment}-as-group"
-  vpc_zone_identifier = var.subnet_ids_gitlab_runner
-
+  name                      = var.enable_forced_updates ? "${var.environment}-as-group" :"${aws_launch_configuration.gitlab_runner_instance.name}-asg"
```

Comment:
I've left the old name as-is for backwards compatibility, but it does mean that the pattern changes depending on if you're using the forced updates. I'd use the full `-as-group`, but I'm worried that the length of the name will become a problem.